clash-prelude/tests/Clash/Tests/FixedExhaustive.hs

```diff
@@ -0,0 +1,203 @@
+     )
+  => Proxy f
+  -> Assertion
+forAllEnumFromThenTo Proxy = sequence_
+  [ let fs = enumFromThenTo x1 x2 y
+        rs = enumFromThenTo (toRational x1) (toRational x2) (toRational y)
+        prefix = unlines [ "x1 = Fixed " ++ show (toInteger $ unFixed x1)
+                         , "x2 = Fixed " ++ show (toInteger $ unFixed x2)
+                         , "y = Fixed " ++ show (toInteger $ unFixed y)]
+    in if (x1 == x2) then
+         listsEqual prefix (take 10 fs) (take 10 rs)
+       else
+         listsEqual prefix fs rs
+  | x1 :: f <- map Fixed [minBound..]
+  , x2 <- map Fixed [minBound..]
+  , y <- map Fixed [minBound..]]
+
+enumTests
+  :: forall f rep int frac
+   . ( FracFixedC rep int frac
+     , Typeable rep
+     , f ~ Fixed rep int frac
+     )
+  => Proxy f
+  -> TestTree
+enumTests pf =
+  testGroup (show $ typeRep pf)
+    [ testCase "enumFrom" $ forAllEnumFrom pf
+    , testCase "enumFromThen" $ forAllEnumFromThen pf
+    , testCase "enumFromTo" $ forAllEnumFromTo pf
+    , testCase "enumFromThenTo" $ forAllEnumFromThenTo pf ]
+
+limitTestWidth
+  :: forall f rep int frac
+   . ( FracFixedC rep int frac
+     , Typeable rep
+     , f ~ Fixed rep int frac
+     )
+  => (Proxy f -> TestTree)
+  -> Proxy f
+  -> Maybe TestTree
+limitTestWidth f pf = if natToNatural @(int + frac) <= widthLimit then
+                        Just (f pf)
+                      else
+                        Nothing
+
+tests :: TestTree
+tests = testGroup "FixedExhaustive" $ catMaybes
+  [ limitTestWidth enumTests (Proxy @(SFixed 0 0))
+  , limitTestWidth enumTests (Proxy @(SFixed 0 1))
+  , limitTestWidth enumTests (Proxy @(SFixed 1 0))
+  , limitTestWidth enumTests (Proxy @(SFixed 0 2))
+  , limitTestWidth enumTests (Proxy @(SFixed 1 1))
+  , limitTestWidth enumTests (Proxy @(SFixed 2 0))
+  , limitTestWidth enumTests (Proxy @(SFixed 0 3))
+  , limitTestWidth enumTests (Proxy @(SFixed 1 2))
+  , limitTestWidth enumTests (Proxy @(SFixed 2 1))
+  , limitTestWidth enumTests (Proxy @(SFixed 3 0))
+  , limitTestWidth enumTests (Proxy @(SFixed 0 4))
+  , limitTestWidth enumTests (Proxy @(SFixed 1 3))
+  , limitTestWidth enumTests (Proxy @(SFixed 2 2))
+  , limitTestWidth enumTests (Proxy @(SFixed 3 1))
+  , limitTestWidth enumTests (Proxy @(SFixed 4 0))
+  , limitTestWidth enumTests (Proxy @(SFixed 0 5))
+  , limitTestWidth enumTests (Proxy @(SFixed 1 4))
+  , limitTestWidth enumTests (Proxy @(SFixed 2 3))
+  , limitTestWidth enumTests (Proxy @(SFixed 3 2))
+  , limitTestWidth enumTests (Proxy @(SFixed 4 1))
+  , limitTestWidth enumTests (Proxy @(SFixed 5 0))
+  , limitTestWidth enumTests (Proxy @(SFixed 0 6))
+  , limitTestWidth enumTests (Proxy @(SFixed 1 5))
+  , limitTestWidth enumTests (Proxy @(SFixed 2 4))
+  , limitTestWidth enumTests (Proxy @(SFixed 3 3))
+  , limitTestWidth enumTests (Proxy @(SFixed 4 2))
+  , limitTestWidth enumTests (Proxy @(SFixed 5 1))
+  , limitTestWidth enumTests (Proxy @(SFixed 6 0))
```

Comment:
I'm axing this.

As martijn said, at a quick glance people might conclude these datatypes are tested when in fact they are excluded through `widthLimit`. And on second thought it doesn't add useful functionality, not useful enough.